### PR TITLE
YD-66 disclose goal conflict url

### PR DESCRIPTION
**Pull request by aristotelos**

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/DisclosureTest.groovy [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+	def 'Richard checks his anonymous messages and finds the disclosure request'(){
+		given:
+
+		when:
+			def response = appService.getAnonymousMessages(richardQuinURL, richardQuinPassword)
+			if(response.status == 200 && response.responseData._embedded.discloseRequestMessages) {
+				goalDiscloseRequestMessage1AcceptURL = response.responseData._embedded.discloseRequestMessages[0]._links.accept.href
+				goalDiscloseRequestMessage2RejectURL = response.responseData._embedded.discloseRequestMessages[1]._links.reject.href
+			}
```
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/DisclosureTest.groovy]: The request should contain a link to the message for which the disclose is requested.

File: core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java [hunk 2]
```diff
@@ -66,10 +65,7 @@ public MessageDestination getAnonymousDestination()
 
 	public Set<MessageDestination> getAllRelatedDestinations()
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java]: Rename into ``getBuddyDestinations``.

File: core/src/main/java/nu/yona/server/messaging/entities/DiscloseRequestMessage.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import nu.yona.server.crypto.Decryptor;
+import nu.yona.server.crypto.Encryptor;
+import nu.yona.server.subscriptions.entities.UserAnonymized;
+
+@Entity
+public class DiscloseRequestMessage extends Message
+{
+	private UUID targetGoalConflictMessageID;
+	private Status status;
+	private String nickname;
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/messaging/entities/DiscloseRequestMessage.java]: Must be encrypted.

File: core/src/main/java/nu/yona/server/messaging/entities/DiscloseResponseMessage.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import nu.yona.server.analysis.entities.GoalConflictMessage.Status;
+import nu.yona.server.crypto.Decryptor;
+import nu.yona.server.crypto.Encryptor;
+
+@Entity
+public class DiscloseResponseMessage extends Message
+{
+	private UUID targetGoalConflictMessageID;
+	private Status status;
+	private String nickname;
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/messaging/entities/DiscloseResponseMessage.java]: Must be encrypted.

File: core/src/main/java/nu/yona/server/messaging/service/DiscloseRequestMessageDTO.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+@JsonRootName("discloseRequestMessage")
+public class DiscloseRequestMessageDTO extends MessageDTO
+{
+	private static final String ACCEPT = "accept";
+	private static final String REJECT = "reject";
+
+	private String nickname;
+	private boolean isAccepted;
+	private boolean isRejected;
+
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/messaging/service/DiscloseRequestMessageDTO.java]: Introduce a status enum instead. Now you have an impossible combination: accepted and rejected.

Comment (aristotelos) [inline core/src/main/java/nu/yona/server/messaging/service/DiscloseRequestMessageDTO.java]: Can be done, I copied this from BuddyConnectRequestMessageDTO. If you like I will refactor both.

File: core/src/main/java/nu/yona/server/messaging/entities/DiscloseResponseMessage.java (hunk 4 above)
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/messaging/entities/DiscloseResponseMessage.java]: Done

File: core/src/main/java/nu/yona/server/messaging/entities/DiscloseRequestMessage.java (hunk 3 above)
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/messaging/entities/DiscloseRequestMessage.java]: Done

File: appservice/src/intTest/groovy/nu/yona/server/DisclosureTest.groovy (hunk 1 above)
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/DisclosureTest.groovy]: Done

File: core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java (hunk 2 above)
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java]: Done

File: core/src/main/java/nu/yona/server/messaging/service/DiscloseRequestMessageDTO.java (hunk 5 above)
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/messaging/service/DiscloseRequestMessageDTO.java]: OK. Created (YD-22)[http://jira.yona.nu/browse/YD-122] for this.

**Pull request merged**